synapse/api/constants.py

```diff
@@ -39,217 +41,217 @@ class Membership:
 
     """Represents the membership states of a user in a room."""
 
-    INVITE = "invite"
-    JOIN = "join"
-    KNOCK = "knock"
-    LEAVE = "leave"
-    BAN = "ban"
-    LIST = (INVITE, JOIN, KNOCK, LEAVE, BAN)
+    INVITE: Final = "invite"
```

Comment:
I was wondering if this should say `Final[str]`. Answer seems to be a big fat NO.
If you make it `Final[str]`, then it types as `str` rather than `Literal['invite']`.
So I think this is fine.